Assets/HoloToolkit/Input/Scripts/InputSources/XboxControllerInputSource.cs

```diff
@@ -0,0 +1,113 @@
+            controllerData.XboxRightStick_Down = Input.GetButtonDown("XBOX_RIGHT_STICK_CLICK");
+            controllerData.XboxView_Down = Input.GetButtonDown("XBOX_VIEW");
+            controllerData.XboxMenu_Down = Input.GetButtonDown("XBOX_MENU");
+
+            controllerData.XboxA_Pressed = Input.GetButton("XBOX_A");
+            controllerData.XboxB_Pressed = Input.GetButton("XBOX_B");
+            controllerData.XboxX_Pressed = Input.GetButton("XBOX_X");
+            controllerData.XboxY_Pressed = Input.GetButton("XBOX_Y");
+            controllerData.XboxLeftBumper_Pressed = Input.GetButton("XBOX_LEFT_BUMPER");
+            controllerData.XboxRightBumper_Pressed = Input.GetButton("XBOX_RIGHT_BUMPER");
+            controllerData.XboxLeftStick_Pressed = Input.GetButton("XBOX_LEFT_STICK_CLICK");
+            controllerData.XboxRightStick_Pressed = Input.GetButton("XBOX_RIGHT_STICK_CLICK");
+            controllerData.XboxView_Pressed = Input.GetButton("XBOX_VIEW");
+            controllerData.XboxMenu_Pressed = Input.GetButton("XBOX_MENU");
+
+            controllerData.XboxA_Up = Input.GetButtonUp("XBOX_A");
+            controllerData.XboxB_Up = Input.GetButtonUp("XBOX_B");
+            controllerData.XboxX_Up = Input.GetButtonUp("XBOX_X");
+            controllerData.XboxY_Up = Input.GetButtonUp("XBOX_Y");
+            controllerData.XboxLeftBumper_Up = Input.GetButtonUp("XBOX_LEFT_BUMPER");
+            controllerData.XboxRightBumper_Up = Input.GetButtonUp("XBOX_RIGHT_BUMPER");
+            controllerData.XboxLeftStick_Up = Input.GetButtonUp("XBOX_LEFT_STICK_CLICK");
+            controllerData.XboxRightStick_Up = Input.GetButtonUp("XBOX_RIGHT_STICK_CLICK");
+            controllerData.XboxView_Up = Input.GetButtonUp("XBOX_VIEW");
+            controllerData.XboxMenu_Up = Input.GetButtonUp("XBOX_MENU");
+
+            InputManager.Instance.RaiseXboxInputUpdate(this, sourceId, controllerData);
+        }
+
+        protected override void RefreshDevices()
+        {
+            var joystickNames = Input.GetJoystickNames();
+
+            if (joystickNames.Length <= 0) { return; }
+
+            if (LastDeviceList != null && (joystickNames.Length != LastDeviceUpdateCount || string.IsNullOrEmpty(joystickNames[0])))
+            {
+                foreach (var gamePadInputSource in gamePadInputDatas)
+                {
+                    InputManager.Instance.RaiseGamePadLost(this, gamePadInputSource.Key, LastDeviceList[gamePadInputSource.Key]);
+                }
+
+                gamePadInputDatas.Clear();
+            }
+
+            for (var i = 0; i < joystickNames.Length; i++)
+            {
+                if (string.IsNullOrEmpty(joystickNames[i]) || gamePadInputDatas.ContainsKey((uint)i)) { continue; }
+
+                if (joystickNames[i].Contains("Xbox Bluetooth Gamepad") || joystickNames[i].Contains("Xbox Wireless Controller") || joystickNames[i].Contains("Xbox Controller"))
```

Comment:
Are there gonna be more of these names? Maybe have a fixed list of these so this line doesn't get any longer.

Reply:
Not exactly a fixed list from Unity side.

The joystick list gets changed every time a new one is recognized by either the player or the editor.

Reply:
I mean a fixed list for the contains names

Reply:
Will do.